src/page/ProductList.page.ts

```diff
@@ -1,11 +1,22 @@
-import { $, ElementFinder, promise } from 'protractor';
+import { $, ElementFinder, ElementArrayFinder, promise, $$ } from 'protractor';
 
 export class ProductListPage {
   private get tShirtImage(): ElementFinder {
     return $('.product_img_link > img');
   }
 
-  public goToShirtDetailsMenu(): promise.Promise<void> {
-    return this.tShirtImage.click();
+  private get productContainerList(): ElementArrayFinder {
+    return $$('.product-container');
   }
+
+  public goToShirtDetailsMenu(productName: string): promise.Promise<void> {
+    return this.selectProduct(productName).$('img').click();
+  }
+
+  private selectProduct(productName: string): ElementFinder {
+    return this.productContainerList
+    .filter(elem => elem.$('.product-name').getText()
+    .then(text => text === productName)).first();
```

Comment:
the method's readability is a bit confusing, maybe some indentation can help

```js
return this.productContainerList
  .filter(elem =>
    elem
      .$('.product-name')
      .getText()
      .then(text => text === productName))
  .first();
```

or (I haven't compiled it you'll have to try 😈 )

```js
return this.productContainerList
  .map(element => element.$('.product-name').getText())
  .filter(elementText =>elementText.then(text => text === productName))
  .first();
```